src/dialects/abstract/query-generator.js

```diff
@@ -2224,17 +2224,22 @@ export class AbstractQueryGenerator extends AbstractQueryGeneratorTypeScript {
     }
   }
 
-  _getSelectOptimizerHints(_options) {
-    return 'SELECT';
+  _getBeforeSelectFragment(_options) {
+    return '';
+  }
+
+  _getAfterSelectFragment(_options) {
+    return '';
   }
 
   selectFromTableFragment(options, model, attributes, tables, mainTableAs) {
     this._throwOnEmptyAttributes(attributes, { modelName: model && model.name, as: mainTableAs });
 
     this._validateSelectOptions(options);
 
-    let fragment = this._getSelectOptimizerHints(options);
-
+    let fragment = this._getBeforeSelectFragment(options);
+    fragment += 'SELECT';
+    fragment += this._getAfterSelectFragment(options);
```

Comment:
I still recommend naming this one something more explicit like `_getBeforeSelectAttributesFragment`. We could need to add more fragments in the future and the current name makes me think it goes at the end of the select query